cfn-resources/cluster/docs/README.md

```diff
@@ -139,6 +141,16 @@ _Allowed Values_: <code>AWS</code> | <code>GCP</code> | <code>AZURE</code> | <co
 
 _Update requires_: [No interruption](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-no-interrupt)
 
+#### GlobalClusterSelfManagedSharding
+
+Flag that indicates if cluster uses Atlas-Managed Sharding (false) or Self-Managed Sharding (true).
+
+_Required_: No
+
+_Type_: Boolean
+
+_Update requires_: [Replacement](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-replacement)
```

Comment:
similar to the concern mentioned in terraform, it would seem that marking property as `createOnly` forces a replacement if modified. Is this intentional? We could consider leaving as a regular property. cc: @lantoli

Reply:
See [comment above](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/1079#discussion_r1645675545)

Reply:
@EspenAlbert I think it's useful to do a manual test of creating a stack and then modifying the param and update the stack to see if we get an error (correct) instead of a cluster replacement (bad)

Reply:
![image](https://github.com/mongodb/mongodbatlas-cloudformation-resources/assets/6596097/8f941070-ee41-4acc-a0e8-c766e0ebc81a)

I switched it from `null` to `false` and got the error message above (no cluster deletion) ☝️ 

Then I did a different update (setting `PitEnabled: false` (was true)) and it worked (see screenshot 👇 )

![image](https://github.com/mongodb/mongodbatlas-cloudformation-resources/assets/6596097/7c2c5ef2-2d10-4437-b0d4-ee8038ea3f27)

Can you confirm this is the correct behavior @lantoli ?

Reply:
Good catch, I wonder if that works as expected or it's an unexpected API behavior? 

Also @EspenAlbert any chance to convert this into a test instead of leaving it manual?

Reply:
@marcosuma 
To me it is expected behavior ✅  but I want to check with @lantoli before merging.
Yes, plan on adding that in a separate ticket, see [discussion above](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/1079#discussion_r1645469457)

Reply:
@EspenAlbert thanks, can you do the last test of starting with attribute true, and then try to change to false and if it fails or the cluster is recreated?

Reply:
@EspenAlbert just to be completely sure (in case null is treated differently), can you do 2 more tests, one creating with false and trying to change to true, and the opposite, starting with true and change to false? thanks!

Reply:
Done ✅ 
Getting the same error in both cases

<img width="1179" alt="image" src="https://github.com/mongodb/mongodbatlas-cloudformation-resources/assets/6596097/9ae7da66-4da9-4b12-b67a-118a200ba010">